lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1402,3 +1425,53 @@ def _get_deployable_vhd_path(
             wait_copy_blob(blob_client, vhd_path, log)
 
         return full_vhd_path
+
+    def _node_generate_min_capability(
+        self, req_cap: schema.NodeSpace, azure_cap: schema.NodeSpace
+    ) -> schema.NodeSpace:
+        assert azure_cap.features, "azure node has no disk features."
+        if req_cap.features is None or (
+            not req_cap.features.__contains__(DiskType.DISK_EPHEMERAL)
+            and not req_cap.features.__contains__(DiskType.DISK_PREMIUM)
+            and not req_cap.features.__contains__(DiskType.DISK_STANDARD)
+        ):
+            # If no disk type feature is specified, use least
+            # cost available disk from azure_cap
+            min_cost_disk = AzurePlatform._get_min_cost_disk(azure_cap.features)
+        else:
+            min_cost_disk = AzurePlatform._get_min_cost_disk(req_cap.features)
+
+        # Remove mutually exclusive feature
+        azure_cap_copy = copy.deepcopy(azure_cap)
+        azure_cap_copy.features.update(  # type: ignore
+            copy.deepcopy(azure_cap.features.items)
+        )
+
+        for me_feature in features.MUTUALLY_EXCLUSIVE_FEATURES[min_cost_disk]:
```

Comment:
It should raise exception, if test case requirement includes conflict features.

Reply:
It's possible for `azure_cap` to have conflicting features. A azure VM could support standard, premium and ephemeral disk. The logic here is trying to remove mutually exclusive features once `min_cost_disk` disk has been set. 

The check you mentioned should be on `req_cap`. I will add it at the beginning of `_node_generate_min_capability`.